mock_clock.go

```diff
@@ -10,30 +10,6 @@ func init() {
 	close(closedChan)
 }
 
-// Clock is a wrapper around common functions in the time package. This interface
-// is designed to allow easy mocking of time functions.
-type Clock interface {
-	// Now returns the current time.
-	Now() time.Time
-
-	// After returns a channel which receives the current time after
-	// the given duration elapses.
-	After(duration time.Duration) <-chan time.Time
-
-	// Sleep blocks until the given duration elapses.
-	Sleep(duration time.Duration)
-
-	// Since returns the time elapsed since t.
-	Since(t time.Time) time.Duration
-
-	// Until returns the duration until t.
-	Until(t time.Time) time.Duration
-
-	// NewTicker will construct a ticker which will continually fire,
-	// pausing for the given duration in between invocations.
-	NewTicker(duration time.Duration) Ticker
-}
-
```

Comment:
I moved the interfaces to their own files because it didn't make sense to me to keep them in the mock files. I had to search a little to find them to do updates originally.